clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -556,8 +556,13 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
   mlir::Value VisitUnaryImag(const UnaryOperator *E) {
     llvm_unreachable("NYI");
   }
+
   mlir::Value VisitUnaryExtension(const UnaryOperator *E) {
-    llvm_unreachable("NYI");
+    TestAndClearIgnoreResultAssign();
```

Comment:
This diverges from the original clang CodeGen. Could you add some explanation regarding this change?

Reply:
I can't, because it is an error. I fixed this